packages/presentation/lib/src/pages/widgets/game_form.dart

```diff
@@ -79,23 +88,24 @@ class Buttons extends StatelessWidget {
       crossAxisAlignment: CrossAxisAlignment.center,
       children: [
         TextButton(
-            onPressed:
-                state is WinState || state is LoseState || state is InitState
-                    ? () {
-                        _formKey.currentState?.reset();
-                        bloc.add(NewGameEvent());
-                      }
-                    : null,
-            child: const Text('New Game')),
+          onPressed: tile.state != BlocTileState.gameInProgress
+              ? () {
+                  _formKey.currentState?.reset();
+                  bloc.add(NewGameEvent());
```

Comment:
u can move key to bloc. in this case u will have the following logic:

onPressed: tile.state != BlocTileState.gameInProgress
               ? () {
                   bloc.launchNewGame()
}

launchNewGame() {
key.currentState?.reset();
// handleNewGame()
}